src/langchain_google_alloydb_pg/vectorstore.py

```diff
@@ -349,6 +351,45 @@ async def aadd_documents(
         ids = await self.aadd_texts(texts, metadatas=metadatas, ids=ids, **kwargs)
         return ids
 
+    def _encode_image(self, uri: str) -> str:
+        """Get base64 string from image URI."""
+        with open(uri, "rb") as image_file:
+            return base64.b64encode(image_file.read()).decode("utf-8")
+
+    async def aadd_images(
+        self,
+        images: List[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Embed images and add to the table."""
+        encoded_images = []
+        if metadatas is None:
+            metadatas = [{"image_uri": uri} for uri in images]
+
+        for uri in images:
+            encoded_image = self._encode_image(uri)
+            encoded_images.append(encoded_image)
+
+        # check if `embed_image()` API is supported by the embedding service used
+        if hasattr(self.embedding_service, "embed_image"):
+            try:
+                embeddings = self.embedding_service.embed_image(images)
+            except Exception as e:
```

Comment:
Is there a more specific Exception we can catch? We are also assuming VertexAI will soon support this, right?

Reply:
There is no more specific exception we can catch because of different embedding implementations. Yes VertexAI will soon be supporting this but there may be future image embedding implementations still not supporting list input.